src/corona_hakab_model/histogram.py

```diff
@@ -0,0 +1,117 @@
+from collections import Counter
+import numpy as np
+from typing import List
+
+from generation.connection_types import ConnectionTypes
+
+
+class Histogram:
+    """
+    This class implements a cumulative normalized histogram, such that each time new data is received,
+    a new normalized histogram is calculated, based on all the accumulated data.
+    The histogram is represented as a dictionary with the following properties:
+     - histogram dict:
+         * 'number_of_connections': The x-axis of the histogram, number of connections for each agent
+         * 'probability_density': The y-axis of the histogram, the probability of an agent the have that number of
+                                  connections.
+         * 'average': The average number of connections for all agents.
+         * 'median': The median number of connections for all agents.
+    """
+    def __init__(self, connections_distribution: List = None):
+        self.weighted_connections = {}
+        self.histogram = {}
+        self.min_connection = 0
+        self.max_connection = 0
+        if connections_distribution is not None:
+            self.update(connections_distribution)
+
+    def __repr__(self):
+        return str(self.histogram)
```

Comment:
Please use pretty printer instead, this representation won't be readable

Reply:
I removed __repr__ and switch to get() instead. I realized the string representation is unnecessary and I return a nested dictionary instead.